src/sdam/topology.ts

```diff
@@ -105,7 +105,8 @@ export interface ServerSelectionRequest {
   mongoLogger: MongoLogger | undefined;
   transaction?: Transaction;
   startTime: number;
-  callback: ServerSelectionCallback;
+  resolve: (server: Server) => void;
+  reject: (error?: AnyError) => void;
   [kCancelled]?: boolean;
```

Comment:
I don't think this should be optional we wouldn't want a promise to reject without an error, if there's anywhere where we may call this with something nullish we should make sure 1. is that actually true or does the TS need fixing 2. if it is true we should make a runtime error that says something about the unexpected nullishness of something.